regparser/builder.py

```diff
@@ -84,16 +84,25 @@ def build_notices(self, checkpoint=True):
         for notice_group in self.eff_notices.values():
             self.notices.extend(notice_group)
 
-    def write_notices(self):
-        for notice in self.notices:
-            #  No need to carry this around
-            del notice['meta']
-            self.writer.notice(notice['document_number']).write(notice)
-
-    def write_regulation(self, reg_tree):
-        self.writer.regulation(self.cfr_part, self.doc_number).write(reg_tree)
-
-    def gen_and_write_layers(self, reg_tree, act_info, cache, notices=None):
+    def write_notices(self, output_type='json'):
+        if output_type == 'json':
+            for notice in self.notices:
+                #  No need to carry this around
+                del notice['meta']
+                self.writer.notice(notice['document_number']).write(notice)
+        elif output_type == 'xml':
+            for notice in self.notices:
+                del notice['meta']
+                self.writer.notice(notice['document_number']).write_notice(notice)
```

Comment:
Is this `if/else` really necessary here? Is it possible to make it so that the builder doesn't care about the output type?